### ISSUE TITLE
Query_string in 5.1 is case sensitive

### ISSUE THREAD

**Issue by hteleb** [elastic/elasticsearch#23620]
I am using AWS ElasticSearch Service for indexing. at first i had version "2.3", when using "query_string for wildcard search it works with upper and lowercase. using same mapping of "2.3" for ES version "5.1" it does not work for uppercase.

ES "2.3"

```
{
    "query": {
        "bool": {
            "must": [{
                "query_string": {
                    "query": "S?ith",
                    "fields": ["authlast"]
                }
            }]
        }
    }
}
```

Explanation result:

```
"explanations": [
    {
      "index": "myindex",
      "valid": true,
      "explanation": "+authlast:s?ith"
    }
  ]
```

upper case "S" in query string mapped to lowercase "s" 

ES :"5.1"
```
"explanations": [
    {
      "index": "myindex",
      "valid": true,
      "explanation": "+authlast:S?ith"
    }
  ]
```
analyzer:
```
"field_analyzer":
{
  "type": "custom",
  "char_filter": "html_strip",
  "tokenizer": "whitespace",
  "filter": [ "lowercase", "asciifolding", "trim" ]
}
```

tried query_string parameters as follows, but still no difference

```
{
    "query": {
        "bool": {
            "must": [{
                "query_string": {
                    "query": "S?ith",
                    "fields": ["authfirst"],
                    "analyzer": "field_analyzer",
                    "analyze_wildcard": true,
                    "lowercase_expanded_terms": "true"
                   
                }
            }]
        }
    }
}
```

**Comment by jimczi** [comment]
Thanks for the report @hteleb 

This is fixed in 5.2 by:
https://github.com/elastic/elasticsearch/pull/21919

Similar problem also reported in https://github.com/elastic/elasticsearch/issues/22540

**Comment by hteleb** [comment]
Thank you Jim for your confirmation. I am afraid AWS does not support 5.2 yet. we may need to  explicitly lowercase the query string before embedding into the "query_string"
